src/Resources/contao/forms/FormCaptcha.php

```diff
@@ -147,27 +147,44 @@ public function validate()
 
 
 	/**
-	 * Generate the captcha question
-	 *
-	 * @return string The question string
+	 * Generate the captcha values and store them in the session
 	 */
-	protected function getQuestion()
+	protected function generateCaptcha()
 	{
+		if ($this->arrCaptcha !== null) {
+			return;
+		}
+
 		$int1 = rand(1, 9);
 		$int2 = rand(1, 9);
 
-		$question = $GLOBALS['TL_LANG']['SEC']['question' . rand(1, 3)];
-		$question = sprintf($question, $int1, $int2);
-
-		/** @var SessionInterface $objSession */
-		$objSession = \System::getContainer()->get('session');
-
-		$objSession->set('captcha_' . $this->strId, array
+		$this->arrCaptcha = array
 		(
+			'int1' => $int1,
+			'int2' => $int2,
 			'sum' => $int1 + $int2,
 			'key' => $this->strCaptchaKey,
 			'time' => time()
-		));
+		);
+
+		/** @var SessionInterface $objSession */
+		$objSession = \System::getContainer()->get('session');
+
+		$objSession->set('captcha_' . $this->strId, $this->arrCaptcha);
+	}
+
+
+	/**
+	 * Generate the captcha question
+	 *
+	 * @return string The question string
+	 */
+	protected function getQuestion()
+	{
+		$this->generateCaptcha();
```

Comment:
I would prefer a more explicit check here:

```php
if (!isset($this->arrCaptcha['int1']) || !isset($this->arrCaptcha['int2']))
{
    $this->generateCaptcha();
}
```

Technically, this is not necessary of course, however, it would regenerate `$this->arrCaptcha` in case the two required properties are not set.